src/OrbitGl/include/OrbitGl/MultivariateTimeSeries.h

```diff
@@ -7,140 +7,62 @@
-  // EndTimeInNs()], return empty array. Otherwise return a range of entries affected by the time
-  // range [min_time, max_time] where:
-  // * the first entry with the time key right before the time range
-  // (min_time, max_time) if exists; otherwise points to the fist entry.
-  // * the last entry with the time key right after the time range
-  // (min_time, max_time) if exists; otherwise points to the last entry.
-  [[nodiscard]] std::vector<std::pair<uint64_t, std::array<double, Dimension>>>
-  GetEntriesAffectedByTimeRange(uint64_t min_time, uint64_t max_time) const {
-    absl::MutexLock lock(&mutex_);
-    if (time_to_series_values_.empty() || min_time >= max_time ||
-        min_time >= time_to_series_values_.rbegin()->first ||
-        max_time <= time_to_series_values_.begin()->first) {
-      return {};
-    }
-
-    auto current_it = GetPreviousOrFirstEntryIterator(min_time);
-    auto last_iterator = GetNextOrLastEntryIterator(max_time);
-
-    std::vector<std::pair<uint64_t, std::array<double, Dimension>>> result;
-    result.push_back(*current_it);
-    do {
-      ++current_it;
-      result.push_back(*current_it);
-    } while (current_it != last_iterator);
-
-    return result;
-  }
+      typename absl::btree_map<uint64_t, std::vector<double>>::const_iterator;
 
  private:
   [[nodiscard]] TimeSeriesEntryIter GetPreviousOrFirstEntryIterator(uint64_t time) const
-      ABSL_EXCLUSIVE_LOCKS_REQUIRED(mutex_) {
-    ORBIT_CHECK(!time_to_series_values_.empty());
-
-    auto iterator_lower = time_to_series_values_.upper_bound(time);
-    if (iterator_lower != time_to_series_values_.begin()) --iterator_lower;
-    return iterator_lower;
-  }
-
+      ABSL_EXCLUSIVE_LOCKS_REQUIRED(mutex_);
   [[nodiscard]] TimeSeriesEntryIter GetNextOrLastEntryIterator(uint64_t time) const
-      ABSL_EXCLUSIVE_LOCKS_REQUIRED(mutex_) {
-    ORBIT_CHECK(!time_to_series_values_.empty());
-
-    auto iterator_higher = time_to_series_values_.lower_bound(time);
-    if (iterator_higher == time_to_series_values_.end()) --iterator_higher;
-    return iterator_higher;
-  }
-
-  void UpdateMinAndMax(double value) ABSL_EXCLUSIVE_LOCKS_REQUIRED(mutex_) {
-    max_ = std::max(max_, value);
-    min_ = std::min(min_, value);
-  }
+      ABSL_EXCLUSIVE_LOCKS_REQUIRED(mutex_);
+  void UpdateMinAndMax(double value) ABSL_EXCLUSIVE_LOCKS_REQUIRED(mutex_);
 
   mutable absl::Mutex mutex_;
-  absl::btree_map<uint64_t, std::array<double, Dimension>> time_to_series_values_
-      ABSL_GUARDED_BY(mutex_);
+  absl::btree_map<uint64_t, std::vector<double>> time_to_series_values_ ABSL_GUARDED_BY(mutex_);
   double min_ ABSL_GUARDED_BY(mutex_) = std::numeric_limits<double>::max();
   double max_ ABSL_GUARDED_BY(mutex_) = std::numeric_limits<double>::lowest();
 
-  const std::array<std::string, Dimension> series_names_;
+  const std::vector<std::string> series_names_;
   const uint8_t value_decimal_digits_;
   const std::string value_unit_;
```

Comment:
Can you remove the `const`  keywords? Members in classes shouldn't be marked const because it disables move-construction.